src/top.rs

```diff
@@ -14,14 +14,16 @@ use std::thread::Builder;
 
 pub struct GeckoMedia {
     sender: Sender<GeckoMediaMsg>,
+    id: usize,
```

Comment:
This doesn't seem right at all to me, given a GeckoMedia isn't supposed to be associated to any media decoder in particular.